koap/build.gradle.kts

```diff
@@ -58,3 +59,19 @@ kotlin {
         }
     }
 }
+
+npmPublishing {
+    organization = "juullabs"
+
+    repositories {
+        repository("github") {
+            registry = uri("https://npm.pkg.github.com")
+        }
+    }
+
+    publications {
+        named("js") {
+            version = project.findProperty("npm.publish.version").toString()
```

Comment:
Instead of having to maintain this, why not just set the version at the top level? (Plugin uses project.version as default)

Reply:
Or you can simply invoke the build from ci just passing -Pversion=xxx and it'll resolve the same with no comfig required in buildfile

Reply:
That way the only thing you need to config is the repository

Reply:
Thanks so much for all the help @mpetuska!

Reply:
FYI, the top level version is set as `unspecified` hence the requirement to inject it:
https://github.com/JuulLabs/koap/blob/dtaylor/apd-954_npm_pkg_on_main_petuska/gradle.properties#L2

Using `-Pversion=xxx` is the ideal solution, thanks for the tip. Will apply this change.

Reply:
Please disregard my reference to the `gradle.properties` `VERSION_NAME` property - that has no bearing here.

Unfortunately the `-Pversion=xxx` option fails to override the `project.version` property, at least when it comes to the value that is emitted into the `package.json` file. It is unclear why this is but I suspect it is to do with the behavior of the Gradle Kotlin DSL configuration?

Will have to stick with `npm.publish.version` at least for now.

Reply:
Apologies, looks like there's a bug in the plugin that's using rootProject.version instead on project.version as default. So yeah, stick with your solution until next release. You could set the version at publication.version instead of publication.packageJson.version to make it more concise

Reply:
> You could set the version at publication.version instead of publication.packageJson.version to make it more concise

I tried this and it didn't work, but that's ok we can wait for the next patch and delete the entire section anyhow.

Reply:
I see. Thanks for letting me know. Something to look into and fix in the next release